markdown_lab/core/errors.py

```
@@ -0,0 +1,327 @@
+# Convenience functions for common error scenarios
+def network_timeout_error(url: str, timeout: int, retry_count: int = 0) -> NetworkError:
+    """Create a network timeout error with standard context."""
+    return NetworkError(
+        f"Request to {url} timed out after {timeout} seconds",
+        url=url,
+        error_code="NETWORK_TIMEOUT",
+        context={"timeout": timeout, "retry_count": retry_count}
+    )
+
+
+def parsing_element_not_found_error(url: str, selector: str, parser_type: str = "html") -> ParsingError:
```

Pylint (3.3.7)

[convention] Line too long (104/100)

(C0301)